src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/FindPal.Windows.cs

```diff
@@ -137,6 +127,29 @@ private unsafe void FindByTime(DateTime dateTime, int compareResult)
 
         public unsafe void FindByTemplateName(string templateName)
         {
+            static unsafe bool DecodeV1TemplateCallback(void* pvDecoded, int cbDecoded, string templateName)
+            {
+                Debug.Assert(cbDecoded >= sizeof(CERT_NAME_VALUE));
+                CERT_NAME_VALUE* pNameValue = (CERT_NAME_VALUE*)pvDecoded;
+                ReadOnlySpan<char> actual = MemoryMarshal.CreateReadOnlySpanFromNullTerminated((char*)pNameValue->Value.pbData);
+                return actual.Equals(templateName, StringComparison.OrdinalIgnoreCase);
+            }
+
+            static unsafe bool DecodeV2TemplateCallback(void* pvDecoded, int cbDecoded, string templateName)
+            {
+                Debug.Assert(cbDecoded >= sizeof(CERT_TEMPLATE_EXT));
+                CERT_TEMPLATE_EXT* pTemplateExt = (CERT_TEMPLATE_EXT*)pvDecoded;
+                string? actual = Marshal.PtrToStringAnsi(pTemplateExt->pszObjId);
```

Comment:
This one could also use `CreateReadOnlySpanFromNullTerminated`, right?

Reply:
I... don't think so but someone can tell me where I can wrong.

`CreateReadOnlySpanFromNullTerminated` has two overloads. The `char*` works for Unicode (UTF-16). The `byte*` works for UTF8 (and thus likely ANSI), but gives you have a `ReadOnlySpan<byte>`, so we'd still need to get it in to a `string` or `ReadOnlySpan<char>` again somehow. So I just left it alone.

Reply:
Ah, that makes sense.  Carry on.